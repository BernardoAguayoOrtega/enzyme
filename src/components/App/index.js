//import react
import React from 'react';

const SubTitle = ({ text }) => <h4>{text}</h4>;

//export and create app component
export const App = () => {
	const handleSomething = () => true;

	return (
		<>
			<h3 className='title'>I'm the app component</h3>
			<h2>Hey there</h2>
			<h3>{name}</h3>
			<SubTitle text='hey' />
		</>
	);
};
