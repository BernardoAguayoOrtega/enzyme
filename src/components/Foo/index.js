import React from 'react';

export function Foo({ name }) {
  return <div className={name} />;
}
