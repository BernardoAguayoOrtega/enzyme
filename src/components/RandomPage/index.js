//import react
import React from 'react';

//create and export random page
export const RandomPage = () => {
	return (
		<>
			<h5>I'm the random page</h5>
			<p>I'm a p tag</p>
		</>
	);
};
