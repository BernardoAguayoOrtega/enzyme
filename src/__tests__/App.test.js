//import react
import React from 'react';
//import app component
import { App } from '../components/App';
//import shallow
import { shallow, mount } from 'enzyme';
//import toJson
import toJson from 'enzyme-to-json';

describe('<App />', () => {
	const wrapper = shallow(<App name='Bernardo' />);

	it('should contain 1 h3 element', () => {
		expect(wrapper.find('h3').exists()).toBe(true);
	});

	it('should contain title class', () => {
		expect(wrapper.find('.title').exists()).toBe(true);
	});

	it('should the correct text', () => {
		expect(wrapper.find('.title').text()).toBe("I'm the app component");
	});

	it('should have only children', () => {
		expect(wrapper.find('.title').children().length).toBe(1);
	});

	it('should bee sub title component', () => {
		expect(wrapper.find('SubTitle[text="hey"]').exists()).toBe(true);
	});

	it;

	it('matches the snap shot', () => {
		expect(toJson(wrapper)).toMatchSnapshot();
	});
});

describe('<App /> mount', () => {
	const wrapper = mount(<App />, );

	it('matches th e snap shot', () => {
		expect(toJson(wrapper)).toMatchSnapshot();
	});
});
