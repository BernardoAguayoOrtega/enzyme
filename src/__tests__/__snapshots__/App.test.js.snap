// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> matches the snap shot 1`] = `
<Fragment>
  <h3
    className="title"
  >
    I'm the app component
  </h3>
  <h2>
    Hey there
  </h2>
  <h3 />
  <SubTitle
    text="hey <3"
  />
</Fragment>
`;
